### PR TITLE
Add actual Gas Price to synthetic transaction receipts

### PR DESCRIPTION
**Pull request by stefan-stefanooov**
**Description**:
This PR modifies **eth_getTransactionReceipt** function in order to add **effectiveGasPrice** for synthetic transactions (HTS), Contract Create or Contract Call.

**Related issue(s)**:  https://github.com/hashgraph/hedera-json-rpc-relay/issues/1736

**Notes for reviewer**:
**Hash** of a test transaction in **testnet**:
0xd624d50e109c51d3765c3ade221797b6003b5b0dd0063c49fad5add8f04ffcc6 

**Checklist**
- [ ] Tested (unit, integration, etc.)
image

### CODE REVIEW

**Review by georgi-l95 — DISMISSED** [review]
LG

File: packages/relay/src/lib/eth.ts [hunk 1]
```diff
@@ -1756,7 +1756,7 @@ export class EthImpl implements Eth {
         blockNumber: cachedLog.blockNumber,
         contractAddress: cachedLog.address,
         cumulativeGasUsed: EthImpl.zeroHex,
-        effectiveGasPrice: EthImpl.zeroHex,
+        effectiveGasPrice: await this.gasPrice(requestIdPrefix),
```
Comment (AlfredoG87) [inline packages/relay/src/lib/eth.ts:1759]: My understanding is that this is the `current` gasPrice, when we should be returning the `gasPrice` at the time of the transaction.

File: package.json [hunk 2]
```diff
@@ -94,10 +95,10 @@
         "!(packages/**/node_modules/)*.ts": [
             "prettier --write"
         ]
-      },
-      "husky": {
+    },
+    "husky": {
         "hooks": {
-          "pre-commit": "lint-staged"
+            "pre-commit": "lint-staged"
```
Comment (AlfredoG87) [inline package.json:101]: why are we adding husky with this PR?

Comment (georgi-l95) [reply package.json:101]: I think this is just indentation fixing from prettier, nothing new added.

**Review by Nana-EC — CHANGES_REQUESTED** [review]
This isn't the approach we want.
As the ticket notes we need to pick up the gasPrice as the time of the transaction.
This is trickier than just taking the current gasPrice.
Happy to sync on this but we should take a different approach of put it back in the backlog and address other tickets

File: packages/relay/src/lib/eth.ts [hunk 3]
```diff
@@ -1750,13 +1750,21 @@ export class EthImpl implements Eth {
       EthImpl.ethGetTransactionReceipt,
       requestIdPrefix,
     );
+
     if (cachedLog) {
+      const block = await this.getBlockByHash(cachedLog.blockHash, false);
+      const gasPriceForTimestamp = await this.getFeeWeibars(
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:1756]: Please add an acceptance test case under `rpc_batch1.spec.ts` that submits an HTS transfer transaction and verifies the transactionReceipt has the expected value

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Nice first addition, please add an acceptance test for an E2E consideration

File: packages/relay/src/lib/eth.ts [hunk 4]
```diff
@@ -1750,13 +1750,21 @@ export class EthImpl implements Eth {
...
+        block?.timestamp,
+      );
+
       const receipt: any = {
         blockHash: cachedLog.blockHash,
         blockNumber: cachedLog.blockNumber,
         contractAddress: cachedLog.address,
         cumulativeGasUsed: EthImpl.zeroHex,
-        effectiveGasPrice: EthImpl.zeroHex,
+        effectiveGasPrice: numberTo0x(gasPriceForTimestamp),
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:1767]: If it's not too much scope creep please consider fixing this for the setting of `effectiveGasPrice` further down should it be null after https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/src/lib/eth.ts#L1794

Then similar ask, add an acceptance tests that checks a submitted ContractCall that executes a transaction and you confirm the `effectiveGasPrice` is correct.
This one should be easier as you can use the details from [mirrorContractDetails.hash](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/server/tests/acceptance/rpc_batch1.spec.ts#L83) to make the query

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Option to fix it all in one go and cover non EthereumTransaction flows also

File: .env.example [hunk 5]
```diff
@@ -4,4 +4,21 @@ MIRROR_NODE_URL=
 HEDERA_NETWORK=
 #mandatory env variables to support queries and transactions to Consensus Node e.g. eth_sendRaTransaction and fallback scenarios on calls to Mirror Node
 OPERATOR_ID_MAIN=<AccountID e.g. 0.1.2, can be obtained from portal.hedera.com, wallet or exchange of choice>
-OPERATOR_KEY_MAIN=<DER Encoded Private Key, can be obtained from portal.hedera.com, wallet or exchange of choice>
\ No newline at end of file
+OPERATOR_KEY_MAIN=<DER Encoded Private Key, can be obtained from portal.hedera.com, wallet or exchange of choice>
```
Comment (Nana-EC) [inline .env.example:7]: Should revert this change

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Exciting change.
Please resolve DCO change and revert `.env` change so we can get this in :)

**Comment by stefan-stefanooov** [comment]
@Nana-EC 
Hello :) 
I have added the two tests you asked for. 
Please have a look and comment on them.
Hopefully all your recommendations are addressed.

P.S.
(the previous push was just a test one. I am sorry for the spam)

**Review by ebadiere — APPROVED** [review]
There are still test failures.

**Review by ebadiere — CHANGES_REQUESTED** [review]
There are still test failures.

File: packages/server/tests/localAcceptance.env [hunk 6]
```diff
@@ -14,3 +14,4 @@ HAPI_CLIENT_TRANSACTION_RESET=50
 HAPI_CLIENT_DURATION_RESET=3600000
 HAPI_CLIENT_ERROR_RESET=[50]
 FILTER_API_ENABLED=true
+ETH_POPULATE_SYNTHETIC_CONTRACT_RESULTS=true
```
Comment (Nana-EC) [inline packages/server/tests/localAcceptance.env:17]: nit: is this not the default? If so why is this needed?

**Review by Nana-EC — APPROVED** [review]

File: packages/server/tests/localAcceptance.env (hunk 6 above)
Comment (stefan-stefanooov) [reply packages/server/tests/localAcceptance.env:17]: I think the default is false. At least that is what is in `~/hedera-json-rpc-relay/docs/configuration.md` And since you asked. I saw in `eth.ts` something strange.

>   private readonly shouldPopulateSyntheticContractResults = process.env.ETH_POPULATE_SYNTHETIC_CONTRACT_RESULTS
    ? process.env.ETH_POPULATE_SYNTHETIC_CONTRACT_RESULTS === 'true'
    : false;
    
Do you think it's a mistake ?

Comment (georgi-l95) [reply packages/server/tests/localAcceptance.env:17]: It's not a mistake, we've decided to keep it false by default and configure it depending on needs, until it's clear that it won't cause any troubles for normal ethereum receipts.
